src/ray/raylet/scheduling/cluster_task_manager.cc

```diff
@@ -163,16 +185,30 @@ void ClusterTaskManager::HandleTaskFinished(std::shared_ptr<WorkerInterface> wor
 }
 
 bool ClusterTaskManager::CancelTask(const TaskID &task_id) {
-  for (auto iter = tasks_to_schedule_.begin(); iter != tasks_to_schedule_.end(); iter++) {
-    if (std::get<0>(*iter).GetTaskSpecification().TaskId() == task_id) {
-      tasks_to_schedule_.erase(iter);
-      return true;
+  for (auto shapes_it = tasks_to_schedule_.begin(); shapes_it != tasks_to_schedule_.end();
```

Comment:
An existing problem, but can we avoid O(n) cancel?

Reply:
We could add a `map<TaskID, Task>` for this?

Reply:
I think we need to have a `<map<TaskID, iterator>>`

Reply:
Couldn't the iterator get invalidated if something else is erased though? Also maybe we should do this in a separate PR?